src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +379,285 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                    }
+                    else
+                    {
+                        // THIS IS A PROBLEM. IF ALL THE "END" PROPERTIES ARE MISSING,
+                        // IT MEANS "THE REST OF THE StartLine". BUT IF CHARLENGTH IS
+                        // PRESENT BUT 0, IT MEANS "INSERTION POINT". AND WE CAN'T TELL
+                        // THE DIFFERENCE.
+                        // TODO: Issue #932
+
+                        // Assume it's an insertion point
+                        region.EndLine = region.EndColumn = region.Length = 0;
+                    }
+                }
+                else
+                {
+                    // There are no text-related properties. Therefore either the region is
+                    // described entirely by binary-related properties, or the region is an
+                    // insertion point at the start of the file.
+                    region.Length = v2Region.ByteLength;
+                    region.Offset = v2Region.ByteOffset;
+                }
             }
 
             return region;
         }
 
+        private int ConvertCharOffsetToByteOffset(int charOffset, Uri uri)
+        {
+            int byteOffset = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    char[] buffer = new char[charOffset];
+
+                    // Read everything up to charOffset
+                    if (reader.ReadBlock(buffer, 0, buffer.Length) > 0)
+                    {
+                        byteOffset = encoding.GetByteCount(buffer, 0, buffer.Length);
+                    }
+                }
+            }
+
+            return byteOffset;
+        }
+
+        private int GetRegionByteLength(Region v2Region, Uri uri)
+        {
+            int byteLength = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    if (v2Region.StartLine > 0) // Use line and column 
+                    {
+                        string sourceLine = string.Empty;
+
+                        // Read down to startLine (null return means EOF)
+                        for (int i = 1; i <= v2Region.StartLine && sourceLine != null; sourceLine = reader.ReadLine(), i++) { }
+
+                        if (sourceLine != null)
+                        {
+                            if (sourceLine.Length > v2Region.StartColumn)
+                            {
+                                // Since we read past startColumn, we need to back up using the base stream
+                                Stream stream = reader.BaseStream;
+                                stream.Position -= encoding.GetByteCount(sourceLine.Substring(v2Region.StartColumn - 1));
```

Comment:
Bug: `StartColumn` might be 0, in which case it defaults to 1.
```
int startColumn = v2Region.StartColumn > 0
    ? v2Region.StartColumn
    : 1;
stream.Position -= encoding.GetByteCount(sourceLine.Substring(startColumn - 1));
```

Reply:
Fixed

Reply:
In fact, you have to calculate that defaulted start column right after the test for `sourceLine != null`, because you use it in the test `sourceLine.length > v2Region.StartColumn`.

Reply:
Two issues here:
1. As I said in my follow-on comment above, you must calculate the defaulted `startColumn` before Line 510, because the test on that line uses `v2Region.StartColumn`.
2. Although you calculated `startColumn`, you didn't actually use in on Line 517.

In summary:
```
if (sourceLine != null)
{
    int startColumn = v2Region.StartColumn > 0
        ? v2Region.StartColumn
        : 1;

    if (sourceLine.Length > startColumn)
    {
        // Since we read past startColumn, we need to back up using the base stream.
        Stream stream = reader.BaseStream;

        stream.Position -= encoding.GetByteCount(sourceLine.Substring(startColumn - 1));
        ...
```

Reply:
Fixed